### DISCUSSION TITLE
更新日志 - 22.03.6 - 20240112

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#1252] [Announcements]
__同步 OpenWRT 22.03.6；FullCone-NAT 高性能模式__

~~部分用户反馈升级以后RTL8125网卡会卡住系统，如果是RTL8125的用户，希望确保升级以后不会出问题，升级前可以在后台页面的系统-杂项设置-驱动里面勾选r8125，然后再升级。如果已经升级了固件，不管有没有遇到问题，都可以到issue里反馈下或者查看解决方案，链接 #1257 更新：已确认是linux内核bug，下次发布时会修复。~~ 使用最新固件即可解决

关于passwall的问题：
1. 从这里下载passwall，以避免openssl版本的问题 https://github.com/bcseputetto/Are-u-ok/releases
2. passwall默认开机60秒以后才开始运行，这是非常不合理的，尤其对于x86设备，到passwall的高级设置里面，把“开机时延时启动”设置成5以内的数字就行了
3. 如果不确定节点是否支持ipv6，那么，先在后台的“网络”-“DHCP/DNS”-“高级设置”，勾选“过滤 IPv6 AAAA 记录”，以避免ipv6转发。
4. 为避免DNS缓存污染，或者DNS对应的ip没加入ipset，客户端（手机电脑等）如果访问外网失败就重新联网试试。最好软路由重启以后，把wifi路由器也重启下，强制所有设备重新联网。
5. 负载均衡启动失败，重装haproxy试试：软件包那边点一下更新列表，然后搜索haproxy，卸载重装。可能之后要重装下passwall。
6. 如果进行以上操作以后还有问题，再来反馈

# 系统
1. 同步 OpenWRT 上游更新 22.03.6，内核升级到 5.10.201
7. FullCone-NAT 支持“高性能模式”（移植自Lean's LEDE），可在防火墙页面开启 #759 #1236
8. 修复部分机型风扇控制的“风扇停止温度”不生效问题，这同时解决风扇频繁启停的问题 #1162
9. 修复“杂项设置”的“启动”页面配置的等待硬盘功能不可靠的问题，并支持等待 RAID 阵列

# 应用
* QuickStart：
  1. 初步支持深色主题
  2. NTFS 早已不再有性能缺陷，所以取消相关提示，转为提示 NTFS，FAT，exFAT 等不支持 Docker，且只在分区列表对话框提示
  3. 修复 Swap 分区提示只读的问题，对于未挂载的 Swap 分区显示格式化按钮 #1204
  4. 分区列表对于未挂载的分区也显示分区大小（非文件系统大小）

# 设备
* x86_64：单网口设备默认DHCP客户端，更符合用户习惯，方便NAS类型，以及VPS云端部署（同时由于单网口默认情况下是LAN口，也就是允许网络访问，注意安全） istoreos/istoreos-cloud#6

**Comment by haobinnan** [comment]
什么时候能编译好？，电犀牛r66s也会在第一波更新吗？

**Reply by jjm2473** [reply]
已经在上传了，估计半小时内能更新完

**Reply by haobinnan** [reply]
回答问题速度真快 哈哈，感谢快速答复！https://github.com/istoreos/istoreos/issues/1254 https://github.com/istoreos/istoreos/issues/1253 这两个问题希望能看下。

**Comment by xixiluhui520** [comment]
老大我更新以后感觉内容是更新了的，但是版本号一直还是老大刷了几次了，刷下载好的包也是这样

**Reply by xixiluhui520** [reply]
怎么刷都是：
iStoreOS 22.03.5 2023122916

**Reply by jjm2473** [reply]
`cat /rom/etc/os-release` 看看 rom 里面的版本是什么

**Reply by xixiluhui520** [reply]
马上看看

**Reply by xixiluhui520** [reply]
是iStoreOS 22.03.6 2024011211老大，我该怎么办

**Reply by xixiluhui520** [reply]
![QQ浏览器截图20240112223105](https://github.com/istoreos/istoreos/assets/139526512/dbef9f79-984f-4b60-93f0-2b3350f17d28)
![Uploading QQ浏览器截图20240112223105.jpg…]()

**Reply by xixiluhui520** [reply]
![QQ浏览器截图20240113102103](https://github.com/istoreos/istoreos/assets/139526512/783fc02e-7b9c-456a-9824-95a1bd78c58c)
![QQ浏览器截图20240113102118](https://github.com/istoreos/istoreos/assets/139526512/89316117-0c2d-4a43-8fab-5586c5498a28)

**Reply by jjm2473** [reply]
固件确实升级了的，但是不知道你怎么修改了overlay的对应文件。要么就恢复出厂设置，要么在/overlay/upper里面找到对应文件删掉，然后重启。但我估计你的系统应该不会只改了这几个文件。

**Reply by docker-master** [reply]
我也有这个问题，请详细讲讲哪个目录，我KVM虚拟机做个旁路由，用替换大法，把好的配置导入到主路由里面

**Reply by docker-master** [reply]
终于看到和我一样的了，我就不害怕了，就这么用着吧，一开始拿来当路由用，现在有了ipv6当服务器用，最重要的是稳定，我要保证尽量不重启

**Comment by hzz-GH** [comment]
更新后passwall用不了了？

**Reply by xixiluhui520** [reply]
你更新后的版本正常吗，我更新后还是老版本号，但是管理面板的主题变了

**Reply by hzz-GH** [reply]
更新后是这个版本号
iStoreOS 22.03.6 2024011211 / LuCI istoreos-22.03 branch git-24.009.14990-dc7527f

**Reply by DR-lin-eng** [reply]
有个依赖没了,要去下个这个libopenssl3,在https://downloads.openwrt.org/releases/23.05.0/packages这里找到对应cpu型号然后base文件夹里面手动安装了就行了

**Reply by tizips** [reply]
Passwall 的负载均衡（Haproxy）一直启动失败

**Reply by jjm2473** [reply]
@tizips 看日志，还有重装haproxy试试，软件包那边点一下更新列表，然后搜索haproxy，卸载重装

**Reply by tizips** [reply]
确实，Haproxy 的依赖有问题，重新安装就好了，3q

**Comment by WONGWAI1017** [comment]
Passwall 2好像挂了

**Reply by hzz-GH** [reply]
我卸载了passwall重新安装好像又可以了

**Reply by hzz-GH** [reply]
但是现在  系统—软件包 里面找不到了  卸载都卸不了

**Reply by jjm2473** [reply]
看看系统日志或者插件自己的日志有没有报错，有的话发出来。理论上小版本更新不会有兼容性问题。
如果你更新以后开了FullCone-NAT的“高性能模式”，也试试改回默认的或者关掉。

**Reply by DR-lin-eng** [reply]
有个依赖没了,要去下个这个libopenssl3,在https://downloads.openwrt.org/releases/23.05.0/packages这里找到对应cpu型号然后base文件夹里面手动安装了就行了

**Reply by mirtu-ko** [reply]
passwall 4.71-2和passwall2 1.16-7都无法使用ssr节点，目前passwall 4.71-2尝试hyteria2节点正常。

尝试安装libopenssl3无效，更新libopenssl-conf到3.12也无效。

**Reply by jjm2473** [reply]
passwall没问题，自己禁用ipv6，并且客户端重新连接网络，以避免dns污染

**Reply by mirtu-ko** [reply]
我这边关闭所有ipv6，依然无法使用。同样的配置把节点从ssr协议改成hyteria2就可以正常使用。对比了一下，发现切到ssr节点，多了类似这样的报错：
dial tcp 127.0.0.1:2082: connect: connection refused

2082应该是分流规则里的。
ssr节点时，好像并没有节点信息：
```
"outbounds": [
    {
      "server_port": 2082,
      "version": "5",
      "udp_over_tcp": false,
      "type": "socks",
      "server": "127.0.0.1",
      "tag": "default"
    },]
```

如果是hyteria2，则是这样
```
"outbounds": [
    {
      "tls": {
        "enabled": true,
        "insecure": true,
        "ech": {
          "enabled": false,
          "pq_signature_schemes_enabled": false,
          "dynamic_record_sizing_disabled": false
        },
        "server_name": "-----"
      },
      "type": "hysteria2",
      "server_port": -----,
      "password": "-----",
      "obfs": [],
      "server": "-----",
      "tag": "default"
    },
    {
      "server_port": 2082,
      "version": "5",
      "udp_over_tcp": false,
      "type": "socks",
      "server": "127.0.0.1",
      "tag": "OpenAI"
    },]
```

**Reply by jjm2473** [reply]
@mirtu-ko 如果是配置文件生成有错误，估计是passwall插件的bug，看看有没有其他版本试试。

**Reply by jjm2473** [reply]
你有试过passwall吗，据我所知，passwall2是passwall的精简版

**Reply by mirtu-ko** [reply]
我有passwall 4.71-2和passwall2 1.16-7两个版本，表现都是一样的，无法使用ssr节点。passwall 4.71-2我尝试过singbox和xray分流，以及不使用分流，目前都无法使用ssr节点。

手上没有其他ssr机场，不过怀疑是节点的兼容问题。

**Reply by bcseputetto** [reply]
@mirtu-ko 这是因为shadowsocksr-libev 依赖从libopenssl1.1变成libopenssl3后，和你机场的ssr配置不兼容了。你可以完全移除shadowsocksr-libev 相关的软件包，然后到 https://github.com/bcseputetto/Are-u-ok/releases 这里下载passwall
如果你搞不定，我建议你重置路由器后再到上述链接，下载并安装passwall

**Reply by mirtu-ko** [reply]
@bcseputetto OK，移除shadowsocksr-libev 相关的软件包然后安装你更新的版本后，问题解决了！感谢！

**Comment by marine19811006** [comment]
还是没敢更新啊，现在这个版本很好，做旁路由可以跑满，就怕升级了就跑不满了

**Comment by wolvth** [comment]
哈哈，终于把连接数限制改了，以前就发现了，默认连接数在一些场景下会爆掉，网络就断了过一会儿才恢复。
也是小白瞎找找到与nf_conntrack有关，半懂不懂的修改后就好了。
https://blog.csdn.net/s_alted/article/details/129192130
主要是这几个文件
除了 /etc/sysctl.d里的几个配置文件还有这些有关
# /proc/sys/net/netfilter/nf_conntrack_buckets
# /sys/module/nf_conntrack/parameters/hashsize
# /proc/sys/net/netfilter/nf_conntrack_max

**Comment by kennylee60** [comment]
这个深色模式怎么可以搞出来

**Reply by xixiluhui520** [reply]
你系统的主题要是深色

**Reply by jxhsjxhs** [reply]
![be4b473bc3dc1e3c4d13cff039c9587](https://github.com/istoreos/istoreos/assets/16410777/601235d2-7706-46ed-830c-37d5462a2e8b)
这里可以改

**Reply by jjm2473** [reply]
Argon主题设置强制暗色或者跟随系统都行，只对默认的argon主题有效，如果你使用其他主题应该会根据系统自动切换。

**Comment by fankangsong** [comment]
国内网站都打不开了，怎么回事

**Reply by jjm2473** [reply]
重启试试。还有你用的是openclash吗？

**Reply by fankangsong** [reply]
我用 passwall，更新后无法使用 ipv4 的解析，默认都是 ipv6解析导致所有的ipv4网站都打不开。重启也没用，最好重置了配置恢复了。重装 passwall 无法工作。

**Reply by DR-lin-eng** [reply]
有个依赖没了,要去下个这个libopenssl3,在https://downloads.openwrt.org/releases/23.05.0/packages这里找到对应cpu型号然后base文件夹里面手动安装了就行了

**Reply by jjm2473** [reply]
不需要安装什么依赖，使用这个仓库编译好的 https://github.com/bcseputetto/Are-u-ok/releases

我已经测试过passwall没问题的。

还有，如果安装以后上不了外网，有可能是因为dns缓存污染，电脑或者手机重新联网，或者清除浏览器和系统的dns缓存就行了。如果使用了ipv6，也自己确认下节点是否支持ipv6，最保险的方式就是先关掉ipv6测试。

**Reply by jjm2473** [reply]
@fankangsong 后台的“网络”-“DHCP/DNS”-“高级设置”，有禁止ipv6解析的功能“过滤 IPv6 AAAA 记录”，如果你不确定自己的节点是否支持ipv6，那就勾选这个，避免麻烦。

**Reply by fankangsong** [reply]
重置多次后，安装 are-u-ok 的 passwall 包可以了。我的宽带支持 ipv6，有公网 ipv6 地址。我升级后遇到的问题没有复现，大概问题我描述下：
可以访问 纯 ipv6 的网站，例如 ipw.cn
所有 ipv4 的网站，都打不开，看上去是默认走了 ipv6 AAAA 的解析，导致解析不到 ip
关闭所有ipv6的也不行，包括 dns，dhcp 等设置。
重置后多次安装 passwall 才能用，有几次 passwall 的 trojan 永远连不上

**Reply by jjm2473** [reply]
@fankangsong 看主贴的说明，安装多次才能用，可能只是因为dns缓存污染而已，跟安装多少次没关系，关键要重新联网。
ipv6有没有影响不需要猜测，使用nslookup，curl等命令测试即可，如果google之类的通过ipv6访问，节点又不支持ipv6，肯定会失败。

**Comment by wolvth** [comment]
开高性能模式后，系统日志内核日志都疯狂报错：nf_conntrack: expectation table full
实际使用暂时没看到影响。开兼容模式后消失
网上查询应该是这个参数的关系 /proc/sys/net/netfilter/nf_conntrack_expect_max

**Reply by jjm2473** [reply]
那你试试执行 
```
sysctl -w net.netfilter.nf_conntrack_buckets=65535; sysctl -w net.netfilter.nf_conntrack_expect_max=65535
```
看还有没有这个错误

**Reply by wolvth** [reply]
我照网上说的修改nf_conntrack_expect_max就好了，buckets没动。
报错时sysctl -a |grep nf_conntrack查询nf_conntrack_expect_max的参数应该是992（具体忘记了）。

**Reply by jjm2473** [reply]
nf_conntrack_expect_max默认应该是16384，你现在改成多少了，能承载多少连接数

**Reply by wolvth** [reply]
我自己修改的是63488，和buckets一样。
升级后默认应该不会是16384，如果是16384就不会报错了。
也不知道为什么会有992这个数值（63488/992刚好64，不太懂，我的机器内存试1G）。
我刚跑去改回1000就又有提示了”nf_conntrack: expectation table full"，改成2000就好了，更极限的数值就没试了。当前活动连接数1.5W左右
我在升级这个版本之前自己修改过连接数262144（63488*4，因为超过会丢包断网“nf_conntrack: table full, dropping packets"），但没有动过这个expect。

**Reply by jjm2473** [reply]
@wolvth 不对啊，默认值应该是：
```
net.netfilter.nf_conntrack_buckets=16384
net.netfilter.nf_conntrack_expect_max=16384
```

你执行下`ls -l /etc/sysctl.d/`看看里面有啥文件

**Reply by wolvth** [reply]
4个文件啊，10-default，11-br，11-nf，12-nf**max。我以前也只改过里面的net.netfilter.nf_conntrack_max=262144，buckets=63488
可能是个例吧，不知道为什么会是992。这个好像也不影响使用，再看有没其他人反馈吧。我跨了好几个版本升的。

**Comment by toplezs** [comment]
When will "Quickstart" be released in English?

**Reply by jjm2473** [reply]
At present, internationalization is not our focus, and human resources are limited. You can try iStoreX (install it from iStore), most of the interface has been translated into English, but the functions are not yet complete.

**Comment by xixiluhui520** [comment]
![QQ浏览器截图20240114143621](https://github.com/istoreos/istoreos/assets/139526512/cdea881e-f98d-4f6e-b4d5-4048bb5129a5)
今天我两个软路由都出现这个情况了，求解决

**Reply by jjm2473** [reply]
如果过几分钟还是检查失败，可能是你的外网没设置好

**Reply by ks-bao** [reply]
昨天边缘服务器出了短暂的问题，已经都 OK 了，自己再试试
